doc/sai-ptf/L3_test_plan.md

```diff
@@ -0,0 +1,333 @@
+2. Change other elements in the packets as well, including ``DIP:192.168.60.1-192.168.60.10`` and ``L4_port``
+3. Verify packets received on different lags and their members
+
+- test_ecmp_hash_field
+
+1. Generate Packets, with different source ip as ``SIP:192.168.0.1-192.168.0.10`` 
+3. For each case, set only one of the four ECMP hash fields(exclude SAI_NATIVE_HASH_FIELD_DST_IP)
+4. For each field just change the corresponding field, for example, for SAI_NATIVE_HASH_FIELD_IP_PROTOCOL, use UDP or TCP
+5. Verify packets received on different lags and their members
+6. Change other un-related fields in the packet
+7. Verify packet received on a certain Lag member, not changed
+
+- test_ecmp_hash_seed
+
+1. Send Packets, with ``SIP:192.168.0.1`` ``DIP:192.168.60.1`` 
+2. Verify the packet received on one of the member ports in LAG1-4
+3. Check and change ecmp_default_hash_seed
+4. Send the same packet as the Step2
+5. Verify the packet received on one of the member ports in LAG1-4 but different from step2
+
+- test_ingress_no_diff
+
+1. Generate Packets, with ``SIP:192.168.0.1`` ``DIP:192.168.60.1``
+2. Send packet from Port5 - Port8
+3. Verify packet received on a certain LAG's member, with corresponding SIP, DIP, and ``SMAC: SWITCH_MAC``
+4. Generate Packets, with ``SIP:192.168.0.1`` ``DIP:192.168.60.2``
+5. Send packet from Port5 - Port8
+6. Verify packet received on a certain LAG's member but different from step4
+
+- test_ingress_disable
+
+1. Generate Packets, with ``SIP:192.168.0.1`` ``DIP:192.168.60.1``
+2. Disable the ingress for Port5 - Port8 related RIF
+3. Send packet from Port5 - Port8
+4. Verify no packet was received on any LAG member
+
+
+- test_lag_ecmp_lpm
+
+1. Generate Packets, with different source ip as ``SIP:192.168.0.1-192.168.0.10`` 
+2. Change other elements in the packets, including ``DIP:192.168.60.1`` ``L4_port``
+3. Verify packet received on different lags and their members, with corresponding SIP, DIP, and SMAC: SWITCH_MAC
+4. Create a next-hop group and add two next hops on IP ``10.1.1.101`` ``10.1.2.101`` on LAG1 and LAG2
+5. Set the Routes on ``DIP:192.168.60.1`` as ``SWITCH_MAC_2``  and bind to the new next-hop group
+6. Generate Packets, ``SIP:192.168.0.1-192.168.0.10`` 
+8. Sent packets with ``DIP:192.168.60.1`` and change the ``L4_port`` number in different packets
+9. Verify Packets only get received on LAG1 and LAG2 and with ``SMAC: SWITCH_MAC_2``
+
+- test_lag_ecmp_lpm_add
+
+1. run steps 1-10 in test_ecmp_lpm
+2. add next hop for ip ``10.1.2.101`` to the next-hop group in test_ecmp_lpm
+3. Generate Packets, with different source ip ``SIP:192.168.0.1-192.168.0.10`` 
+4. Change other elements in the packets, including ``DIP:192.168.60.1`` ``L4_port``
+5. Verify Packets can be received on LAG1 LAG2 and LAG3, with ``SMAC: SWITCH_MAC_2``
+
+- test_lag_ecmp_lpm_disable
+
+1. run steps 1-8 in test_ecmp_lpm_add
+2. Disable LAG3 members
```

Comment:
how to disable lag3, disable the port or remove it from next hop group?

Reply:
a member attribute, added more notes